src/adapter/src/config/sync.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use tokio::time;
+
+use crate::config::{
+    SynchronizedParameters, SystemParameterBackend, SystemParameterFrontend,
+    SystemParameterSyncConfig,
+};
+use crate::Client;
+
+/// Run a loop that periodically pulls system parameters defined in the
+/// LaunchDarkly-backed [SystemParameterFrontend] and pushes modified values to the
+/// `ALTER SYSTEM`-backed [SystemParameterBackend].
+pub async fn system_parameter_sync(
+    sync_config: SystemParameterSyncConfig,
+    adapter_client: Client,
+    tick_interval: Option<Duration>,
+) -> Result<(), anyhow::Error> {
+    let Some(tick_interval) = tick_interval else {
+        tracing::info!("skipping system parameter sync as tick_interval = None");
+        return Ok(());
+    };
+
+    // Ensure the frontend client is initialized.
+    let mut frontend = Option::<SystemParameterFrontend>::None; // lazy initialize the frontend below
+    let mut backend = SystemParameterBackend::new(adapter_client).await?;
+
+    // Tick every `tick_duration` ms, skipping missed ticks.
+    let mut interval = time::interval(tick_interval);
+    interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+
+    // Run the synchronization loop.
+    tracing::info!(
+        "synchronizing system parameter values every {} seconds",
+        tick_interval.as_secs()
+    );
+
+    let mut params = SynchronizedParameters::default();
+    loop {
+        // Wait for the next sync period
+        interval.tick().await;
+
+        // Fetch current parameter values from the backend
+        backend.pull(&mut params).await;
+
+        if !params.enable_launchdarkly() {
+            if frontend.is_some() {
+                tracing::info!("stopping system parameter frontend");
+                frontend = None;
+            }
```

Comment:
I think we should have either a debug or info line here that indicates the sync loop is running but the sync itself has been explicitly disabled. Just to help with debugging in production.

```suggestion
            } else {
                tracing::info!("system parameter sync is disabled; not syncing")
            }
```

Reply:
```rust
tracing::info!("system parameter sync is disabled; not syncing")
```

I think with the current config this line will appear ~2x per minute. Are we OK with such level of noise when the sync is off? Alternatively we can set this as a `tracing::debug` line and then explicitly tweak the `log_filter` variable to see that.